File: 748-largest-number-at-least-twice-of-others/largest-number-at-least-twice-of-others.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        m1=0;m2=0;ind=0
        for i in range(len(nums)):
            if nums[i]>m1:
                m2=m1
                m1=nums[i]
                ind=i
            elif nums[i]>m2:
                m2=nums[i]
        return ind if m1>=2*m2 else -1

File: 748-largest-number-at-least-twice-of-others/largest-number-at-least-twice-of-others.cs
public class Solution {
    public int DominantIndex(int[] nums) {
        int max1=0,max2=0,index=0;
        for(int i=0; i<nums.Length; i++){
            if(nums[i]>max1){
                max2=max1;
                max1=nums[i];
                index=i;
            }
            else if(nums[i]>max2){
                max2=nums[i];
            }
        }
        return max1>=2*max2 ? index : -1;
    }
}